src/hidimstat/base_perturbation.py

```diff
@@ -0,0 +1,190 @@
+import numpy as np
+import pandas as pd
+from joblib import Parallel, delayed
+from sklearn.base import BaseEstimator, check_is_fitted
+from sklearn.metrics import root_mean_squared_error
+
+from hidimstat.utils import _check_vim_predict_method
+
+
+class BasePerturbation(BaseEstimator):
+    def __init__(
+        self,
+        estimator,
+        loss: callable = root_mean_squared_error,
+        n_permutations: int = 50,
+        method: str = "predict",
+        n_jobs: int = 1,
+    ):
+        """
+        Base class for model-agnostic variable importance measures based on
+        perturbation.
+
+        Parameters
+        ----------
+        estimator : sklearn compatible estimator, optional
+            The estimator to use for the prediction.
+        loss : callable, default=root_mean_squared_error
+            The function to compute the loss when comparing the perturbed model to the
+            original model.
+        n_permutations : int, default=50
+            This parameter is relevant only for PermutationImportance or CPI.
+            Specifies the number of times the variable group (residual for CPI) is
+            permuted. For each permutation, the perturbed model's loss is calculated and
+            averaged over all permutations.
+        method : str, default="predict"
+            The method used for making predictions. This determines the predictions
+            passed to the loss function. Supported methods are "predict",
+            "predict_proba", "decision_function", "transform".
+        n_jobs : int, default=1
+            The number of parallel jobs to run. Parallelization is done over the
+            variables or groups of variables.
+        """
+        check_is_fitted(estimator)
+        self.estimator = estimator
+        self.loss = loss
+        _check_vim_predict_method(method)
+        self.method = method
```

Comment:
Is the entered value checked somewhere ?

Reply:
yes, if not the method is not supported, it raises an error. Would you have done it differently?